diff --git a/Sources/pfd/pfd_codec_flac.c b/Sources/pfd/pfd_codec_flac.c
index 24007ec..0552c6e 100644
--- a/Sources/pfd/pfd_codec_flac.c
+++ b/Sources/pfd/pfd_codec_flac.c
@@ -28,6 +28,8 @@
 #include "pfd_def.h"
 #include "pfd_private.h"
 
+#ifdef TANGO_USE_FLAC_DECODER
+
 #include "FLAC/all.h"
 
 /*
@@ -465,3 +467,4 @@ void Pfd_Flac_FillFileInfo(PFD_FLAC_HANDLE *handle, PFD_fileinfo_t *desc)
     desc->ChannelCount = 2; // forced to stereo at the moment
 }
 
+#endif // TANGO_USE_FLAC_DECODER
diff --git a/Sources/pfd/pfd_codec_heaac.c b/Sources/pfd/pfd_codec_heaac.c
index cc6798f..c2eea97 100644
--- a/Sources/pfd/pfd_codec_heaac.c
+++ b/Sources/pfd/pfd_codec_heaac.c
@@ -28,9 +28,9 @@
 #include "pfd_private.h"
 
 #include "MMP_Sequencer.h"
-#include "MMP_Decoder_heaac.h"
 
 #ifdef TANGO_USE_AAC_DECODER
+#include "MMP_Decoder_heaac.h"
 #include "TELPLAYERWRAPPER.h"
 #endif
 
@@ -184,7 +184,7 @@ static void Pfd_Heaac_Cache_Read(Pfd_Heaac_Cache *c, int8_t *buffer, int64_t off
             }
 
             //PFD_TRACE("Read %d from file\n", s);
-            int32_t r = Pfd_CurrentSettings.Callbacks.Read(dest,sizeof(int8),s,Pfd_CurrentSettings.CurrentSong->Stream);
+            int32_t r = Pfd_CurrentSettings.Callbacks.Read(dest,sizeof(int8_t),s,Pfd_CurrentSettings.CurrentSong->Stream);
             if (r < s)
             {
                 // couldn't read all, FAIL
@@ -222,6 +222,7 @@ static void Pfd_Heaac_Cache_Read(Pfd_Heaac_Cache *c, int8_t *buffer, int64_t off
     Pfd_CurrentSettings.CurrentSong->Handler.Aac->seekPosition = o + *size;
 }
 
+#ifdef TANGO_USE_AAC_DECODER
 void Pfd_Heaac_GetData(int8* pBuffer, uint64 fileoffset, uint32 *pBytesRead)
 {
     int64_t offset = (int64_t) fileoffset;
@@ -240,6 +241,7 @@ void Pfd_Heaac_GetData(int8* pBuffer, uint64 fileoffset, uint32 *pBytesRead)
 
     //PFD_TRACE("Returned %d\n", read);
 }
+#endif // TANGO_USE_AAC_DECODER
 
 PFD_ERR Pfd_Heaac_ParseInit(PFD_SONGDESC * Desc)
 {
diff --git a/Sources/pfd/pfd_types.h b/Sources/pfd/pfd_types.h
index 8fc2c49..d85e279 100644
--- a/Sources/pfd/pfd_types.h
+++ b/Sources/pfd/pfd_types.h
@@ -19,7 +19,10 @@
 #define _PFD_TYPES_H_
 
 #include "wav_api.h"
-#include "FLAC/stream_decoder.h"
+
+#ifdef TANGO_USE_FLAC_DECODER
+#  include "FLAC/stream_decoder.h"
+#endif // TANGO_USE_FLAC_DECODER
 
 typedef U32 PFD_STATUS;
 
@@ -96,7 +99,11 @@ typedef struct {
 
 typedef struct
 {
+#ifdef TANGO_USE_FLAC_DECODER
     FLAC__StreamDecoder *decoder;
+#else // !TANGO_USE_FLAC_DECODER
+    void *decoder;
+#endif // !TANGO_USE_FLAC_DECODER
     int32_t min_blocksize;
     int32_t max_blocksize;
 
diff --git a/Sources/wav/wav_api.c b/Sources/wav/wav_api.c
index 697bdd7..b90868c 100644
--- a/Sources/wav/wav_api.c
+++ b/Sources/wav/wav_api.c
@@ -22,8 +22,6 @@
 #include "assert.h"
 
 #include "parrotOS.h"
-#include "Tango/Tango.h"
-#include "tools_debug.h"
 
 #include "wav_api.h"
 
@@ -204,7 +202,7 @@ WAVContext* wav_init(unsigned char* Buffer,unsigned int Size)
 
     if (endianRet == WAV_ENDIAN_ERROR)
     {
-        TANGO_ERROR("WAV Init : Couldn't parse endianness chunk\n");
+        PLOGE(TANGO, "WAV Init : Couldn't parse endianness chunk\n");
         return NULL;
     }
     else
@@ -300,7 +298,7 @@ WAVContext* wav_init(unsigned char* Buffer,unsigned int Size)
         ctx = (WAVContext *) calloc(1, sizeof(WAVContext));
         if (!ctx)
         {
-            TANGO_ERROR("WAV parsing : can't allocate WAV structure\n");
+            PLOGE(TANGO, "WAV parsing : can't allocate WAV structure\n");
             return NULL;
         }
         ctx->CodecTag = compression;
@@ -324,7 +322,7 @@ int wav_header_size(unsigned char* Buffer,unsigned int Size)
 
     if (endianRet == WAV_ENDIAN_ERROR)
     {
-        TANGO_ERROR("WAV Init : Couldn't parse endianness chunk\n");
+        PLOGE(TANGO, "WAV Init : Couldn't parse endianness chunk\n");
         return 0;
     }
     else
@@ -399,8 +397,8 @@ int wav_decode(WAVContext* Handle, unsigned char* Inbuf, unsigned int Bilen, sho
     int32_t bytes_per_in_frame;
     uint8_t *inbuf_start = Inbuf;
 
-    TANGO_ASSERT(*Bolen >= 0, "Output size isn't positive");
-    TANGO_ASSERT(!(*Bolen & 3), "Output size isn't multiple of 4 (16bits ST)");
+    PAL_ASSERT_STR(*Bolen >= 0, "Output size isn't positive");
+    PAL_ASSERT_STR(!(*Bolen & 3), "Output size isn't multiple of 4 (16bits ST)");
     if (!Handle || Handle->CodecTag!=WAV_COMPR_PCM || Handle->Channels>2 || !Handle->Channels ||
         (Handle->BitsPerSample!=16 && Handle->BitsPerSample!=24 && Handle->BitsPerSample!=32)) return -1;
 
@@ -517,7 +515,7 @@ int wav_decode(WAVContext* Handle, unsigned char* Inbuf, unsigned int Bilen, sho
     }
     else
     {
-        TANGO_ASSERT(0, "Wrong Endianness");
+        PAL_ASSERT_STR(0, "Wrong Endianness");
     }
     *Bolen = framesToCpyFromInput << 2;
 
