diff --git a/SoftAT/Build/ConfigCK5050Set.in b/SoftAT/Build/ConfigCK5050Set.in
index 699f1c9..a40fe48 100644
--- a/SoftAT/Build/ConfigCK5050Set.in
+++ b/SoftAT/Build/ConfigCK5050Set.in
@@ -305,13 +305,11 @@ menu "CK5050 Software Support"
 	
 				if CK5050_SUPPORT_NETWORK_DEVICE
 					config CK5050_SUPPORT_NADI_DONGLE_3G
-						depends on PAL_PPP_DRIVER
 						depends on BUILD_MODMAN
 						bool "Dongle 3G"
 						default n
 
 					config CK5050_SUPPORT_NADI_DUN
-						depends on PAL_PPP_DRIVER
 						bool "DUN"
 						default n
 						if CK5050_SUPPORT_NADI_DUN
diff --git a/SoftAT/atom.mk b/SoftAT/atom.mk
index 9cf2f00..e4c5e37 100644
--- a/SoftAT/atom.mk
+++ b/SoftAT/atom.mk
@@ -10,7 +10,9 @@ include $(CLEAR_VARS)
 LOCAL_MODULE := softat
 LOCAL_PBUILD_HOOK := 1
 
-LOCAL_CONFIG_FILES := Build/ConfigCK5050Set.in
+LOCAL_CONFIG_FILES := \
+	softat-alchemy-config.in \
+	Build/ConfigCK5050Set.in
 $(call load-config)
 
 SOFTAT_BUILD_DIR := $(call local-get-build-dir)
