diff --git a/src/modman.c b/src/modman.c
index 60e40cb..4bc5f66 100755
--- a/src/modman.c
+++ b/src/modman.c
@@ -55,11 +55,12 @@ struct driver_cfg {
 };
 
 static const struct driver_cfg const drivers[] = {
+
 	{"hso", "/sys/bus/usb/drivers/hso"},
 	{"option", "/sys/bus/usb/drivers/option"},
 	{"sierra", "/sys/bus/usb/drivers/sierra"},
 	{"cdc_acm", "/sys/bus/usb/drivers/cdc_acm"},
-	{"blackberry", "/sys/bus/usb/drivers/blackberry"},
+//	{"blackberry", "/sys/bus/usb/drivers/blackberry"},
 	{"ppp_generic", "/sys/class/ppp"},
 	{NULL, NULL},
 };
