diff --git a/SoftAT/Build/ConfigCK5050Set.in b/SoftAT/Build/ConfigCK5050Set.in
index 699f1c9..a197f3b 100644
--- a/SoftAT/Build/ConfigCK5050Set.in
+++ b/SoftAT/Build/ConfigCK5050Set.in
@@ -1,6 +1,44 @@
 
 ####################################### CK5050 Software SUPPORT ######################################
 menu "CK5050 Software Support"
+
+
+	config BUILD_BLUES
+		bool
+		default y
+
+	config BUILD_TANGO
+		bool
+		default y
+
+	config BUILD_TAMTAM
+		bool
+		default y
+
+	config BUILD_TALA
+		bool
+		default y
+
+	config BUILD_DISCO
+		bool
+		default y
+
+	config BUILD_DISCO_WITH_USB
+		bool
+		default y
+
+	config BUILD_CONCERTOS
+		bool
+		default y
+
+	config BUILD_TANGO
+		bool
+		default y
+
+	config BUILD_RAP
+		bool
+		default y
+
 	# hidden options depending on selected packages
 	config CK5050_SUPPORT_BLUETOOTH
 		bool
@@ -241,7 +279,6 @@ menu "CK5050 Software Support"
 				default y
 	
 			config CK5050_SUPPORT_PAN
-				select CK5050_ENABLE_BNEP_BLOCK
 				bool "CK5050_SUPPORT_PAN"
 				default y
 			if CK5050_SUPPORT_PAN
@@ -305,13 +342,10 @@ menu "CK5050 Software Support"
 	
 				if CK5050_SUPPORT_NETWORK_DEVICE
 					config CK5050_SUPPORT_NADI_DONGLE_3G
-						depends on PAL_PPP_DRIVER
-						depends on BUILD_MODMAN
 						bool "Dongle 3G"
 						default n
 
 					config CK5050_SUPPORT_NADI_DUN
-						depends on PAL_PPP_DRIVER
 						bool "DUN"
 						default n
 						if CK5050_SUPPORT_NADI_DUN
@@ -331,7 +365,6 @@ menu "CK5050 Software Support"
 							"Management and configuration for RNDIS Host network interface."
 
 					config CK5050_SUPPORT_NADI_GSM_MODEM
-						depends on BUILD_REGGAE
 						bool "Gsm Modem"
 						default n
 
@@ -358,7 +391,6 @@ menu "CK5050 Software Support"
 
 		menu "Network Interface"
 			config CK5050_SUPPORT_NETWORK_INTERFACE
-			depends on BUILD_TINYXML
 				bool "CK5050_SUPPORT_NETWORK_INTERFACE"
 				default n
 				help
@@ -474,7 +506,6 @@ menu "CK5050 Software Support"
         config CK5050_SUPPORT_DEVI_IPOD
             bool "CK5050_SUPPORT_DEVI_IPOD"
             default y
-            select BUILD_ACID
             help
                 "If activated, the host will be notified of plugged apple devices.
                 WARNING : This must be activated if you register IPod sources in Concertos."
@@ -498,7 +529,6 @@ menu "CK5050 Software Support"
         config CK5050_SUPPORT_DEVI_OTHER
             bool "CK5050_SUPPORT_DEVI_OTHER"
             default y
-            select PAL_JUBA_CLIENT_SERVICE
             help
                 "If activated, the host will be notified of plugged devices other than IPods (such as USB keys, SD cards, etc ...)."
 
@@ -508,7 +538,6 @@ menu "CK5050 Software Support"
 
     	config CK5050_SUPPORT_UPNP
             bool "CK5050_SUPPORT_UPNP"
-            depends on BUILD_UPNP
 	        select SOFTAT_USE_ASYNCHRONOUS_FILE_LISTING
             help
                 "Allow to browse Upnp servers, and then to perform downloads and streaming from them. To enable it check Libraries/Connectivity/Upnp library.
diff --git a/SoftAT/Export/config.h b/SoftAT/Export/config.h
index 3101052..aed5584 100644
--- a/SoftAT/Export/config.h
+++ b/SoftAT/Export/config.h
@@ -14,6 +14,25 @@
 #ifndef __CONFIG_H__
 #define __CONFIG_H__
 
+#ifdef ALCHEMY_BUILD
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+// add parrot os headers
+#include "parrotOS_define_c.h"
+#include "parrotOS_supervis.h"
+#include "parrotOS_uart_rt.h"
+#include "parrotOS_log.h"
+
+#ifdef __cplusplus
+}
+#endif
+
+#include "blues-config.h"
+
+#else // !ALCHEMY_BUILD
+
 //==================================================================================================
 // Common Definitions.
 //==================================================================================================
@@ -182,5 +201,7 @@ extern "C" {
 }
 #endif // __cplusplus
 
+#endif // !ALCHEMY_BUILD
+
 #endif // __CONFIG_H__
 
