diff --git a/Sources/MediaSources/Java/JavaSourceList.cpp b/Sources/MediaSources/Java/JavaSourceList.cpp
index 677f45b..132e81e 100644
--- a/Sources/MediaSources/Java/JavaSourceList.cpp
+++ b/Sources/MediaSources/Java/JavaSourceList.cpp
@@ -216,6 +216,7 @@ public:
 							 Concertos_MediaSource_Cnx connectivity, Concertos_MediaSource_Features features,
 							 bool connected)
 	{
+#ifdef CONCERTOS_USE_JAVA_SOURCE
 		Concertos_Java_Parameters params;
 		Concertos_Java_InitParameters(&params);
 		params.serviceId = serviceId;
@@ -227,6 +228,7 @@ public:
 		params.features = features;
 		params.connected = connected;
 		Concertos_Java_RegisterEx(m_instance, &params, m_config);
+#endif // CONCERTOS_USE_JAVA_SOURCE
 	}
 
 private:
