diff --git a/utils/iptables/atom.mk b/utils/iptables/atom.mk
index 7ed980b..3dff4f4 100644
--- a/utils/iptables/atom.mk
+++ b/utils/iptables/atom.mk
@@ -10,7 +10,8 @@ LOCAL_AUTOTOOLS_ARCHIVE := $(LOCAL_MODULE)-$(LOCAL_AUTOTOOLS_VERSION).tar.bz2
 LOCAL_AUTOTOOLS_DIR := $(LOCAL_MODULE)-$(LOCAL_AUTOTOOLS_VERSION)
 
 LOCAL_AUTOTOOLS_PATCHES := \
-	lucie/iptables-1.4.11.1.patch
+	lucie/iptables-1.4.11.1.patch \
+	lucie/iptables-1.4.11.1-64bit.patch
 
 include $(BUILD_AUTOTOOLS)
 
