diff --git a/src/netbox/netbox_worker.c b/src/netbox/netbox_worker.c
index de9a261..ea635db 100644
--- a/src/netbox/netbox_worker.c
+++ b/src/netbox/netbox_worker.c
@@ -236,7 +236,7 @@ out:
 	return NULL;
 }
 
-inline int nb_worker_get_source(nb_worker_t *worker, nb_source_t **source)
+/*inline*/ int nb_worker_get_source(nb_worker_t *worker, nb_source_t **source)
 {
 	if (NULL == worker || NULL == source)
 		return -1;
@@ -246,7 +246,7 @@ inline int nb_worker_get_source(nb_worker_t *worker, nb_source_t **source)
 	return 0;
 }
 
-inline int nb_worker_get_client_src(nb_worker_t *worker,
+/*inline*/ int nb_worker_get_client_src(nb_worker_t *worker,
 		nb_source_t **client_src)
 {
 	if (NULL == worker || NULL == client_src)
diff --git a/src/netbox/netbox_worker.h b/src/netbox/netbox_worker.h
index e000f96..dd15396 100644
--- a/src/netbox/netbox_worker.h
+++ b/src/netbox/netbox_worker.h
@@ -38,7 +38,7 @@ nb_worker_t *nb_worker_new(nb_source_t *client_src, unsigned cid,
  * @param source In output, source used by the worker
  * @return -1 in case of error , otherwise 0
  */
-inline int nb_worker_get_source(nb_worker_t *worker, nb_source_t **source);
+/*inline*/ int nb_worker_get_source(nb_worker_t *worker, nb_source_t **source);
 
 /**
  * Returns the source of the client which launched the worker
@@ -46,7 +46,7 @@ inline int nb_worker_get_source(nb_worker_t *worker, nb_source_t **source);
  * @param client In output, source of the client which has launched the worker
  * @return -1 in case of error , otherwise 0
  */
-inline int nb_worker_get_client_src(nb_worker_t *worker, nb_source_t **client);
+/*inline*/ int nb_worker_get_client_src(nb_worker_t *worker, nb_source_t **client);
 
 /**
  * Returns the ID of the command, the worker is treating
